From 28dcb6178ffbdbb11f3cd118795ac9988675d441 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sun, 27 Oct 2019 16:08:20 +0100
Subject: [PATCH] Flush standard error after each request

Otherwise log messages created by http.server, e.g. through log_request,
are not displayed if the output of noprs is written to a pipe. Noticed
this while working on an OpenRC service where the output is piped to a
logger.
---
 noprs.hy | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/noprs.hy b/noprs.hy
index 9778040..a555779 100644
--- a/noprs.hy
+++ b/noprs.hy
@@ -44,7 +44,10 @@
         (let [body (.read self.rfile con-len)]
           (if (.from-github? self (.get self.headers "X-Hub-Signature") body)
             (.dispatch-event self body)
-            (.send-error self 403 "HMAC digest validation failed")))))))
+            (.send-error self 403 "HMAC digest validation failed")))))
+
+    ;; Flush stderr, containing log messages created by http.server
+    (sys.stderr.flush)))
 
 (defmacro setg [name value]
   `(do
